nautobot_golden_config/models.py

```diff
@@ -237,8 +292,18 @@ class ComplianceRule(PrimaryModel):  # pylint: disable=too-many-ancestors
         "match_config",
         "config_type",
         "custom_compliance",
+        "config_remediation",
     ]
 
+    @property
+    def remediation_setting(self):
+        """Returns remediation settings for a particular platform."""
+        try:
+            remediation_setting = RemediationSetting.objects.get(platform=self.platform)
+        except RemediationSetting.DoesNotExist as err:
+            raise ValidationError(f"Platform {self.platform.slug} has no Remediation Settings defined.") from err
+        return remediation_setting
```

Comment:
```suggestion
        return RemediationSetting.objects.filter(platform=self.platform).first()
```

Suggestion to address @mzbroch's comment.

Though this will probably break line 431 below, but so would raising a Validation error so not sure which one we actually want here.
```
        if not FUNC_MAPPER.get(self.rule.remediation_setting.remediation_type):
            raise ValidationError(
                f"Remediation {self.rule.remediation_setting.remediation_type} has no associated function set."
            )
        remediation_config = FUNC_MAPPER[self.rule.remediation_setting.remediation_type](obj=self)
```

Reply:
so I'm also not following how remediation_type would ever not be in FUNC_MAPPER considering its a choicefield where no null/blank is not allowed.  This validation error should never have been possible to hit.

Reply:
Gave my interpretation in my PR on using your suggestion, and updating the validation part.